File: 리모컨.py
a, b = map(int,input().split())
cha = abs(a-b)
count = cha // 10
cha %= 10

if cha == 1 or cha == 5 or cha == 10:
    count += 1
elif cha == 2 or cha == 4 or cha == 6 or cha == 9:
    count += 2
elif cha == 3 or cha == 7 or cha == 8:
    count += 3

print(count)

File: 백준/Gold/1107. 리모컨/리모컨.js
const filePath = process.platform === 'linux' ? '/dev/stdin' : './Javascript/input.txt';
const input = require('fs').readFileSync(filePath).toString().trim().split('\n');
const N = +input[0];
const M = +input[1];
const broken = M ? input[2].split(' ') : [];
const maxChannel = N * 2 + 100;
let count = Math.abs(N - 100); // 버튼 누른 횟수

for (let i = 0; i <= maxChannel; i++) {
    const numStr = i.toString();
    const strLen = numStr.length;
    let isValid = true;

    for (let j = 0; j < strLen; j++) {
        // 지금 누른 버튼중에 고장난 버튼이 포함되어 있다면
        if (broken.includes(numStr[j])) {
            isValid = false;
            break;
        }
    }
    if (isValid) {
        count = Math.min(count, strLen + Math.abs(N - i));
    }
}

console.log(count);
